Core.ApplicationServices/HelpTexts/HelpTextApplicationService.cs

```diff
@@ -0,0 +1,121 @@
+                                    Title = parameters.Title,
+                                    Key = parameters.Key
+                                };
+
+                                _helpTextsRepository.Insert(newHelpText);
+                                _helpTextsRepository.Save();
+                                _domainEvents.Raise(new EntityCreatedEvent<HelpText>(newHelpText));
+
+                                return Result<HelpText, OperationError>.Success(newHelpText);
+                            }));
+        }
+
+        public Maybe<OperationError> DeleteHelpText(string key)
+        {
+            return GetHelpTextWithGlobalAdminRights(key, $"User is not allowed to delete help text with key {key}")
+                        .Match(helpText => {
+                            _helpTextsRepository.Delete(helpText);
+                            _helpTextsRepository.Save();
+                            _domainEvents.Raise(new EntityBeingDeletedEvent<HelpText>(helpText));
+                            return Maybe<OperationError>.None; },
+                            error => error);
+                        
+        }
+
+        public Result<HelpText, OperationError> PatchHelpText(string key, HelpTextUpdateParameters parameters)
+        {
+            return GetHelpTextWithGlobalAdminRights(key, $"User is not allowed to patch help text with key {key}")
+                .Bind(existing => PerformUpdates(existing, parameters)
+                    .Bind(updated =>
+                    {
+                        _helpTextsRepository.Update(updated);
+                        _helpTextsRepository.Save();
+                        _domainEvents.Raise(new EntityUpdatedEvent<HelpText>(updated));
+                        return Result<HelpText, OperationError>.Success(updated);
+                    }));
+        }
+
+        private Result<HelpText, OperationError> GetHelpTextWithGlobalAdminRights(string key, string errorMessage)
+        {
+            return WithGlobalAdminRights(errorMessage)
+                .Match(error => error,
+                    () => GetHelpText(key));
+        }
+
+        private Result<HelpText, OperationError> PerformUpdates(HelpText helpText, HelpTextUpdateParameters parameters)
+        {
+            return helpText
+                .WithOptionalUpdate(parameters.Title, (ht, title) => ht.UpdateTitle(title))
+                .Bind(ht => ht.WithOptionalUpdate(parameters.Description,
+                    (ht, description) => ht.UpdateDescription(description)));
+        }
+
+        private Maybe<OperationError> WithAvailableKey(string key)
+        {
+            return _helpTextService.IsAvailableKey(key)
+                ? Maybe<OperationError>.None 
+                : Maybe<OperationError>.Some(new OperationError($"A help text with the provided key { key } already exists", OperationFailure.Conflict));
+
+        }
+
+        private Maybe<OperationError> WithGlobalAdminRights(string errorMessage)
```

Comment:
add a forbindden to the parameter name (forbiddenErrorMessage) so it's clean right away for what error case the message is needed